tools/webpack/blocks.js

```diff
@@ -278,5 +286,12 @@ module.exports = [
 				},
 			],
 		},
+		plugins: [
+			...plugins,
```

Comment:
Integrates webpack plugins used with all other configs:

- Integration with bundle analyzer.
- Integration with the plugin that replaces globals like `process.env.IS_GUTENBERG_PLUGIN`, `process.env.IS_WORDPRESS_CORE`. Once https://github.com/WordPress/gutenberg/pull/50122 lands, it would also make it possible using `SCRIPT_DEBUG` - that could be a way to run `console.log` only in development mode in a way where it's completely removed in the production build.
- Custom setup for `DependencyExtractionWebpackPlugin`, that together with another WP specific plugin `ReadableJsAssetsWebpackPlugin` ensure that all necessary files are created in the filesystem:
<img width="330" alt="Screenshot 2023-05-08 at 21 04 19" src="https://user-images.githubusercontent.com/699132/236910855-7ca6a66e-cb64-4ebc-85fb-31eb9e11b0b8.png">